src/Domain/Donation.php

```diff
@@ -291,9 +293,62 @@ class Donation extends SalesforceWriteProxy
      */
     protected $fundingWithdrawals;
 
-    public function __construct()
+    /**
+     * @param string $amount
+     * @deprecated but retained for now as used in old test classes. Not recommend for continued use - either use
+     * fromApiModel or create a new named constructor that takes required data for your use case.
+     */
+    public static function emptyTestDonation(string $amount, PaymentMethodType $paymentMethodType = PaymentMethodType::Card): self
+    {
+        return new self($amount, 'GBP', $paymentMethodType);
+    }
+
+    private function __construct(string $amount, string $currencyCode, PaymentMethodType $paymentMethodType)
     {
         $this->fundingWithdrawals = new ArrayCollection();
+        $this->setCurrencyCode($currencyCode);
+        $this->setAmountAndMethod($amount, $paymentMethodType);
+    }
+
+    public static function fromApiModel(DonationCreate $donationData, Campaign $campaign): Donation
+    {
+
+        $psp = $donationData->psp;
+        assert($psp === 'stripe');
+
+        $donation = new self($donationData->donationAmount, $donationData->currencyCode, $donationData->paymentMethodType);
+
+        $donation->setPsp($psp);
+        $donation->setDonationStatus(DonationStatus::Pending);
+        $donation->setUuid(Uuid::uuid4());
```

Comment:
Previously we had the more complex `$donation->setUuid((new UuidGenerator())->generateId($this->getEntityManager(), $donation));` but all that was doing in practice was calling `Uuid::uuid4()` for us in more polymorphic way.